tiledb/sm/subarray/relevant_fragments.h

```diff
@@ -94,15 +107,75 @@ class RelevantFragments {
     size_t i_;
   };
 
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+
   /**
-   * Constructs a relevant fragment object for an array with `frag_num`
+   * Constructs a relevant fragments object for an array with `frag_num`
    * fragments.
    */
   RelevantFragments(size_t frag_num)
       : frag_num_(frag_num)
       , relevant_fragments_computed_(false) {
   }
 
+  /**
+   * Constructs a relevant fragments object from a fragment bytemaps
+   * structure.
+   */
+  RelevantFragments(
+      dimension_size_type dim_num,
+      size_t frag_num,
+      std::vector<std::vector<uint8_t>> fragment_bytemaps)
+      : frag_num_(frag_num)
+      , relevant_fragments_computed_(true) {
+    // Recalculate relevant fragments.
+    computed_relevant_fragments_.reserve(frag_num);
+    for (unsigned f = 0; f < frag_num; ++f) {
+      bool relevant = true;
+      for (uint32_t d = 0; d < dim_num; ++d) {
+        if (fragment_bytemaps[d][f] == 0) {
+          relevant = false;
+          break;
+        }
+      }
+
+      if (relevant) {
+        computed_relevant_fragments_.emplace_back(f);
+      }
+    }
+  }
+
+  /**
+   * Constructs a relevant fragments object from another filtering fragments
+   * that are not in [min, max[.
+   */
+  RelevantFragments(
+      RelevantFragments& relevant_fragments, unsigned min, unsigned max)
+      : frag_num_(relevant_fragments.frag_num_)
+      , relevant_fragments_computed_(true) {
+    computed_relevant_fragments_.reserve(relevant_fragments.size());
+    for (auto f : relevant_fragments) {
+      if (f >= min && f < max) {
+        computed_relevant_fragments_.emplace_back(f);
+      }
+    }
+  }
+
+  /**
+   * Constructs a relevant fragments object from a vector.
+   */
+  RelevantFragments(std::vector<unsigned>& relevant_fragments)
+      : frag_num_(0)
```

Comment:
Is 0 correct here? When/how would it be updated to the correct value?

Reply:
Hopefully the new name for the variable clarifies things? It's not used at all when relevant fragments are computed.